contracts/lib/LibLiquidation.sol

```diff
@@ -129,29 +135,25 @@ library LibLiquidation {
 
             // The difference in how much was expected vs how much liquidator actually got.
             // i.e. The amount lost by liquidator
+            // todo this can probably be further simplified
             uint256 amountToReturn = 0;
             uint256 percentSlippage = 0;
             if (avgPrice < receipt.price && receipt.liquidationSide) {
-                amountToReturn = uint256(amountExpectedFor - amountSoldFor);
-                if (amountToReturn <= 0) {
-                    return 0;
-                }
-                percentSlippage =
-                    (amountToReturn * PERCENT_PRECISION) /
-                    amountExpectedFor;
+                amountToReturn = amountExpectedFor - amountSoldFor;
             } else if (avgPrice > receipt.price && !receipt.liquidationSide) {
-                amountToReturn = uint256(amountSoldFor - amountExpectedFor);
-                if (amountToReturn <= 0) {
-                    return 0;
-                }
-                percentSlippage =
-                    (amountToReturn * PERCENT_PRECISION) /
-                    amountExpectedFor;
+                amountToReturn = amountSoldFor - amountExpectedFor;
+            }
+            if (amountToReturn <= 0) {
+                return 0;
             }
+            // multiply by 100 as we expect this to be a percent as an integer eg 50% = 50
+            percentSlippage =
+                PRBMathUD60x18.div(amountToReturn, amountExpectedFor) *
+                100;
```

Comment:
Is this valid? Why doesn't `100` need to be WAD'd also?

Reply:
Easiest way to think about this is in regular integer math. `amountToReturn / amountExpectedFor` returns a percentage as a decimal representation (eg `1/2 = 0.5` however we provide max slippage as an integer representation of the max slippage (eg 50% max slipapge). So to make sure the check for `percentSlippage > maxSlippage` is valid, we do this multiplication. Note we could also just make maxSlippage a decimal representation if we wanted.

We then divide the amount to return by 100 as when you calculate an actual percent value, you want to use the decimal representation (eg 50% of 5 = 0.5 * 5 = 2.5)